src/main/java/org/ethereum/beacon/discovery/task/DiscoveryTaskManager.java

```diff
@@ -96,14 +100,38 @@ private CompletableFuture<Collection<NodeRecord>> performSearchForNewPeers() {
         .execute();
   }
 
-  private CompletableFuture<Collection<NodeRecord>> findNodes(
-      final NodeRecord nodeRecord, final int targetDistance) {
+  @VisibleForTesting
+  static List<Integer> lookupDistances(int targetDistance) {
+    checkArgument(
+        targetDistance >= KBuckets.MINIMUM_BUCKET && targetDistance <= KBuckets.MAXIMUM_BUCKET,
+        "invalid target distance: %s",
+        targetDistance);
+
     final List<Integer> distances = new ArrayList<>();
-    for (int distance = targetDistance;
-        distance <= Math.min(KBuckets.MAXIMUM_BUCKET, targetDistance + 3);
-        distance++) {
-      distances.add(distance);
+    distances.add(targetDistance);
+
+    // We want to prioritize higher distances in the request over lower ones
+    // because lower distances are less likely to contain nodes.
+    for (int offset = 1; targetDistance + offset <= KBuckets.MAXIMUM_BUCKET; offset++) {
+      if (distances.size() >= LOOKUP_REQUEST_LIMIT) {
+        break;
+      }
```

Comment:
nit: I think I'd simplify this to:
```suggestion
    for (int offset = 1; targetDistance + offset <= KBuckets.MAXIMUM_BUCKET && distances.size() < LOOKUP_REQUEST_LIMIT; offset++) {
```

Reply:
Good call. I've updated those.